snippets/snippets.json

```diff
@@ -1,12 +1,172 @@
+		"prefix": "exec",
+		"description": "Create a desktop entry `Exec`",
+		"body": "Exec=${1:command}"
+	},
+	"A desktop entry working directory where application runs in": {
+		"prefix": "path",
+		"description": "Create a desktop entry `Path`",
+		"body": "Path=${1:/path/to/directory}"
+	},
+	"Specify whether run application in a terminal window": {
+		"prefix": "terminal",
+		"description": "Create a desktop entry `Terminal`",
+		"body": "Terminal=${1|true,false|}"
+	},
+	"A desktop entry actions": {
+		"prefix": "actions",
+		"description": "Create a desktop entry `Actions`",
+		"body": "Actions=${1:action1 action2 ...}"
+	},
+	"A desktop entry MIME types": {
+		"prefix": "mimetype",
+		"description": "Create a desktop entry `MimeType`",
+		"body": "MimeType=${1:type1 type1 ...}"
+	},
+	"A desktop entry menu categories": {
+		"prefix": "categories",
+		"description": "Create a desktop entry `Categories`",
+		"body": "Categories=${1:category1 category2 ...}"
+	},
+	"A desktop entry implemented interfaces": {
+		"prefix": "implements",
+		"description": "Create a desktop entry `Implements`",
+		"body": "Implements=${1:interface1 interface2 ...}"
+	},
+	"A desktop entry keywords": {
+		"prefix": "keywords",
+		"description": "Create a desktop entry `Keywords`",
+		"body": "Keywords=${1:keyword1 keyword2 ...}"
+	},
+	"A desktop entry keywords for a specific language": {
+		"prefix": "keywords",
+		"description": "Create a desktop entry `Keywords[...]`",
+		"body": "Keywords[${1:language}_${2:country}]=${3:keyword1 keyword2 ...}"
+	},
+	"Specify whether send a 'remove' message when started with the DESKTOP_STARTUP_ID environment variable set": {
+		"prefix": "startupnotify",
+		"description": "Create a desktop entry `StartupNotify`",
+		"body": "StartupNotify=${1|true,false|}"
+	},
+	"A desktop entry startup WM class": {
+		"prefix": "startupwmclass",
+		"description": "Create a desktop entry `StartupWMClass`",
+		"body": "StartupWMClass=${1:value}"
+	},
+	"A desktop entry url": {
+		"prefix": "url",
+		"description": "Create a desktop entry `URL`\nRequired when: Type = Link",
+		"body": "URL=${1:value}"
+	},
+	"Specify whether prefer run on a more powerful discrete GPU if available": {
+		"prefix": "prefersnondefaultgpu",
+		"description": "Create a desktop entry `PrefersNonDefaultGPU`",
+		"body": "PrefersNonDefaultGPU=${1|true,false|}"
+	},
+	"Specify whether application is a single window": {
+		"prefix": "singlemainwindow",
+		"description": "Create a desktop entry `SingleMainWindow`",
+		"body": "SingleMainWindow=${1|true,false|}"
+	},
+	"A comment": {
+		"prefix": "#",
+		"description": "Create a comment",
+		"body": "# ${1}"
+	},
+	"A region": {
 		"prefix": "region",
-		"description": "Start a folding region",
-		"body": "#region"
-	},
-	"End a region": {
-		"prefix": "endregion",
-		"description": "End a folding region",
-		"body": "#endregion"
-	}
+		"description": "Create a region",
+		"body": "#region\n${1}\n#endregion"
```

Comment:
Simplifying the region snippets is a great idea. I think it can be improved by using

```json
"body": "#region\n${TM_SELECTED_TEXT}${0}\n#endregion"
```

as it allows users to wrap entire sections without too much hastle. ([Snippet reference](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variables))

Reply:
Done 🚀